addon/-private/functional/modifier.ts

```diff
@@ -89,26 +154,28 @@ export default function modifier(
   fn: (
     element: Element,
     positional: unknown[],
-    named: Record<string, undefined>
-  ) => void | Teardown
+    named: object
+  ) => void | Teardown,
+  options?: { eager: boolean }
 ): FunctionBasedModifier<{
   Element: Element;
   Args: {
-    Named: Record<string, unknown>;
+    Named: object;
     Positional: unknown[];
   };
 }> {
   // SAFETY: the cast here is a *lie*, but it is a useful one. The actual return
   // type of `setModifierManager` today is `void`; we pretend it actually
   // returns an opaque `Modifier` type so that we can provide a result from this
   // type which is useful to TS-aware tooling (e.g. Glint).
+  const isEager = !options || options?.eager;
```

Comment:
This const should be moved _up_ above the `SAFETY` note just above; the note is talking about the line just below the `const` (the `return`) and having hte const between them confused me initially 😛

Reply:
Ahhh, heh, whoops, yes. Thank you!